tables/TaQL/ExprNodeSetOpt.cc

```diff
@@ -0,0 +1,450 @@
+      Bool rightC = set[index[0]]->isRightClosed();
+      rownr_t rownr = index[0];
+      // Loop through the next intervals and combine if possible.
+      for (size_t i=1; i<index.size(); i++) {
+        Int64 inx = index[i];
+        T st2 = stvals[inx];
+        T end2 = endvals[inx];
+        // Combine intervals if they overlap.
+        // They do if the next interval starts before end of this one
+        // or if starting at the end and one side of the interval is closed.
+        if (st2 < endval  ||
+            (st2 == endval  &&
+             (set[inx]->isLeftClosed() || set[index[i-1]]->isRightClosed()))) {
+          // Overlap; update end if higher.
+          if (end2 > endval) {
+            endval = end2;
+            rightC = set[inx]->isRightClosed();
+          }
+        } else {
+          // No overlap, so create the interval found and start a new one.
+          newStart.push_back (stval);
+          newEnd.push_back (endval);
+          newLeftC.push_back (leftC);
+          newRightC.push_back (rightC);
+          rowNrs.push_back (rownr);
+          stval  = st2;
+          endval = end2;
+          leftC  = set[inx]->isLeftClosed();
+          rightC = set[inx]->isRightClosed();
+          rownr  = inx;
+        }
+      }
+      // Create the last interval.
+      newStart.push_back (stval);
+      newEnd.push_back (endval);
+      newLeftC.push_back (leftC);
+      newRightC.push_back (rightC);
+      rowNrs.push_back (rownr);
+    }
+    // Now create the correct object.
+    return createOptSet (set, newStart, newEnd, newLeftC, newRightC);
+  }
+
+  template<typename T>
+  TENShPtr TableExprNodeSetOptContSet<T>::createOptSet
+  (const TableExprNodeSet& set,
+   const std::vector<T>& start, const std::vector<T>& end, 
+   const std::vector<Bool>& leftC, const std::vector<Bool>& rightC)
+   {
+    // See if all intervals have the same left and right closedness.
+    // If so, a better version can be used that does not need to test on it
+    // for each compare.
+    Bool same = True;
+    for (size_t i=1; i<leftC.size(); ++i) {
+      if (leftC[i] != leftC[0]  ||  rightC[i] != rightC[0]) {
+        same = False;
+        break;
+      }
+    }
+    // Create the appropriate object.
+    TableExprNodeSetOptContSetBase* optSet;
+    if (same) {
+      if (leftC[0]) {
+        if (rightC[0]) {
+          optSet = new TableExprNodeSetOptContSetCC<T> (set, start, end);
+        } else {
+          optSet = new TableExprNodeSetOptContSetCO<T> (set, start, end);
+        }
+      } else {
+        if (rightC[0]) {
+          optSet = new TableExprNodeSetOptContSetOC<T> (set, start, end);
+        } else {
+          optSet = new TableExprNodeSetOptContSetOO<T> (set, start, end);
+        }
+      }
+    } else {
+      optSet = new TableExprNodeSetOptContSet<T> (set, start, end,
+                                                  leftC, rightC);
+    }
+    TENShPtr ptr(optSet);
+    ///optSet->setRows (rowNrs);
```

Comment:
Is this needed or not?

Reply:
Removed the line as well as all statements using rowNrs. It was not needed anymore.